tests/drivers/i2s/i2s_api/src/test_i2s_errors.c

```diff
@@ -0,0 +1,161 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <zephyr/kernel.h>
+#include <zephyr/ztest.h>
+#include <zephyr/drivers/i2s.h>
+#include "i2s_api_test.h"
+
+#define INVALID_TRIGGER_SETTING 7
+
+ZTEST_USER(i2s_errors, test_i2s_improper_configuration)
+{
+	int err;
+	struct i2s_config invalid_config = { .word_size = 16U,
+					     .channels = 2U,
+					     .format = I2S_FMT_DATA_FORMAT_I2S,
+					     .frame_clk_freq = FRAME_CLK_FREQ,
+					     .block_size = BLOCK_SIZE,
+					     .timeout = TIMEOUT,
+					     .options = I2S_OPT_FRAME_CLK_MASTER |
+							I2S_OPT_BIT_CLK_MASTER,
+					     .mem_slab = &tx_mem_slab };
+
+
+	invalid_config.format =
+		I2S_FMT_DATA_FORMAT_LEFT_JUSTIFIED | I2S_FMT_DATA_FORMAT_RIGHT_JUSTIFIED;
+
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(
+		err, 0,
+		"I2S configuration did not detect improper data format (I2S_FMT_DATA_FORMAT_LEFT_JUSTIFIED | I2S_FMT_DATA_FORMAT_RIGHT_JUSTIFIED)");
+
+	invalid_config.format = I2S_FMT_DATA_FORMAT_I2S | I2S_FMT_DATA_ORDER_LSB;
+
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(
+		err, 0,
+		"I2S configuration did not detect improper stream format (I2S_FMT_DATA_ORDER_LSB)");
+
+	invalid_config.format = I2S_FMT_DATA_FORMAT_I2S;
+	invalid_config.channels = 3U;
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(err, 0,
+			  "I2S configuration did not detect improper channels configuration (3)");
+
+	invalid_config.channels = 2U;
+	invalid_config.options = I2S_OPT_LOOPBACK | I2S_OPT_PINGPONG;
+	err = i2s_configure(dev_i2s, I2S_DIR_TX, &invalid_config);
+	zassert_not_equal(
+		err, 0,
+		"I2S configuration did not detect improper options (I2S_OPT_LOOPBACK | I2S_OPT_PINGPONG)");
```

Comment:
Why this combination of options should be actually considered invalid?

Reply:
Will be removed.